crates/sui-rpc/src/client/v2/objects.rs

```diff
@@ -0,0 +1,98 @@
+use futures::stream::Stream;
+use futures::stream::{self};
+
+use crate::client::v2::Client;
+use crate::client::v2::Result;
+use crate::proto::sui::rpc::v2::ListOwnedObjectsRequest;
+use crate::proto::sui::rpc::v2::Object;
+
+impl Client {
+    /// Creates a stream of objects based on the provided request.
+    ///
+    /// The stream handles pagination automatically by using the page_token from responses
+    /// to fetch subsequent pages. The original request's page_token is used as the starting point.
+    ///
+    /// # Arguments
+    /// * `request` - The initial `ListOwnedObjectsRequest` with search criteria
+    ///
+    /// # Returns
+    /// A stream that yields `Result<Object>` instances. If any RPC call fails, the
+    /// tonic::Status from that request is returned.
+    pub fn list_owned_objects(
+        &self,
+        request: impl tonic::IntoRequest<ListOwnedObjectsRequest>,
+    ) -> impl Stream<Item = Result<Object>> + 'static {
+        let client = self.clone();
+        let request = request.into_request();
+
+        // Extract headers and extensions from the original request
+        let headers = request.metadata().clone();
+        let extensions = request.extensions().clone();
+        let request_inner = request.into_inner();
```

Comment:
We can use [into_parts](https://docs.rs/tonic/latest/tonic/struct.Request.html#method.into_parts) here